plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/GlueIcebergTableOperations.java

```diff
@@ -108,7 +109,20 @@ protected void commitNewTable(TableMetadata metadata)
         CreateTableRequest createTableRequest = new CreateTableRequest()
                 .withDatabaseName(database)
                 .withTableInput(tableInput);
-        stats.getCreateTable().call(() -> glueClient.createTable(createTableRequest));
+        try {
+            stats.getCreateTable().call(() -> glueClient.createTable(createTableRequest));
+        }
+        catch (AlreadyExistsException
+               | EntityNotFoundException
+               | InvalidInputException
+               | ResourceNumberLimitExceededException e) {
+            // clean up metadata files corresponding to the current transaction
+            fileIo.deleteFile(newMetadataLocation);
+            throw e;
+        }
+        catch (RuntimeException e) {
+            throw e;
+        }
```

Comment:
Redundant right?

(i think this could be found by a static code analysis...)

Reply:
Yes, will remove it.